CRM/Contribute/Form/Contribution/Main.php

```diff
@@ -407,6 +407,10 @@ public function buildQuickForm() {
 
     $config = CRM_Core_Config::singleton();
 
+    $contactID = $this->getContactID();
```

Comment:
What happens in case of anonymous user?  If there is no value for $contactID, next 2 lines should not be called.